ccan/compiler.h

```diff
@@ -0,0 +1,289 @@
+/* CC0 (Public domain) - see LICENSE file for details */
+#ifndef CCAN_COMPILER_H
+#define CCAN_COMPILER_H
+#include "config.h"
+
+#ifndef COLD
+#if HAVE_ATTRIBUTE_COLD
```

Comment:
We can't just copy this, we don't have the configure mechanisms to make it work. This just just be flattened to assume gcc without all the #ifdefery

Reply:
By assume gcc you mean assume the attribute exists?
Does it depend on the gcc version?

Reply:
We already require a new enough gcc to have all these ancient attributes. No need to check for them explicitly.